### ISSUE TITLE
AWS::Serverless:Api Tags are not visible in the API Gateway console nder "Configure Tags"

### ISSUE THREAD

**Issue by lskrajny** [awslabs/serverless-application-model#1239]
**Description:**

**Steps to reproduce the issue:**
1.
2.
3.

**Observed result:**

**Expected result:**

**Comment by keetonian** [comment]
@lskrajny it looks like we haven't configured the output `AWS::ApiGateway::RestApi` resource to have those tags. 

Changes to support this:
Add the tags field here: https://github.com/awslabs/serverless-application-model/blob/develop/samtranslator/model/apigateway.py#L23

Add it to the RestApi resource: https://github.com/awslabs/serverless-application-model/blob/develop/samtranslator/model/api/api_generator.py#L87

(Example of adding them to the stage: https://github.com/awslabs/serverless-application-model/blob/develop/samtranslator/model/api/api_generator.py#L212-L213)

Update tests.

**Comment by lskrajny** [comment]
Sent a PR, let me know if all good

**Comment by ShreyaGangishetty** [comment]
@lskrajny there is a merge conflict in the PR. Could you please resolve it and update the PR?

**Comment by keetonian** [comment]
I reviewed your PR and tried deploying a template with the changes (details in #1244), and found that `Tags` were not added to the ApiGateway RestApi resource (but were successfully added to the ApiGateway Stage resource, like is currently implemented in SAM).

I discussed adding tags to both the V1 RestApi and V2 Api resource with ApiGateway and asked why it didn't work. Here are the main takeaways:

1. Adding tags to the RestApi/Api resource when using OpenApi/Swagger to define an API is **not currently supported** on either V1 or V2 APIs.
  1. V1: template deploys, but, like I found today, the tags are not added to the resource.
  1. V2: template does not deploy and fails with an explicit error. This is the desired behavior
1. There is a concept of `tags` in the OpenApi spec. ApiGateway *may* add support for adding tags in the OpenApi document. If this happens, we can likely support tags on both V1/V2 resources. This is pending a deeper dive from their team to see if this is possible.

I'll keep this thread updated as we chase this down. In the meantime, it looks like it isn't yet possible in SAM to support adding tags to the AWS::ApiGateway::RestApi resource.

**Comment by Slooz** [comment]
@keetonian, you're right that for V1, tags aren't added to the resource upon initial deployment. However, if you change the tag and update the stack, the tag is added to the resource. Obviously, this is still a bug, and I reported this behavior to AWS.

But does this mean that adding tags to the RestApi/Api resource when using OpenApi/Swagger to define an API is supported for V1? Is there simply a bug where CloudFormation fails to deploy the initial tag?

**Comment by praneetap** [comment]
@Slooz Support for V2 to add tags in openapi is coming soon in SAM.

**Comment by dkasyanov** [comment]
I see that both V1 and V2 APIs have ability to add tags to the Api and RestApi.
Is there any possibility that SAM will also have functionality to add tags toe Api (not the Stage)?

**Comment by zz** [comment]
Also can't tag Api in SAM 1.0.0. only stage tagged.

**Comment by godrose** [comment]
This is a showstopper for us. Any ETA?

**Comment by bengwall** [comment]
Also wondering if we can get an update as to when this will be fixed.

**Comment by siliconsheep** [comment]
Hi! Still seems to be an issue today, any ETA on a fix / workaround?

**Comment by rdok** [comment]
This issue still persists as of SAM CLI version 1.36.0. 

**However**, auto generated tags from AWS SAM are actually DO properly work. 
This is important, because these tags cover the majority of uses cases including bill cost tags & least priveleges IAM policies.

_For the auto generated tags to work, [tags/*](https://github.com/rdok/aws-sam-ts-graphql/blob/4efe32f84832842130a97f920d0aa9d727480f3d/cicd-authorisation/lib/api-gateway-policy.ts#L15) resources must be authorised._

![2022-01-02 05_01_11-API Gateway — Mozilla Firefox](https://user-images.githubusercontent.com/4212119/147866757-59d4157a-05e4-45f0-a9ab-15b18c138853.png)

**Comment by ehicks71** [comment]
Any progress on this? I'm still seeing this as an issue.
